### ISSUE TITLE
p-pickList Accessibility Issues

### ISSUE THREAD

**Issue by srujana10499** [primefaces/primeng#18770]
### Describe the bug

The p-picklist component with checkboxes has accessibility violations when running the IBM Equal Accessibility Checker. The issue is: aria specification violations

### Reason for not contributing a PR

- [x] Lack of time
- [ ] Unsure how to implement the fix/feature
- [ ] Difficulty understanding the codebase
- [ ] Other

### Reproducer

https://stackblitz.com/edit/stackblitz-starters-ycuctrbo

### Environment

Stackblitz

### Angular version

20.0.4

### PrimeNG version

v20

### Steps to reproduce the behavior

1. Install the [IBM Equal Accessibility Checker](https://chromewebstore.google.com/detail/ibm-equal-access-accessib/lkcagbfjnkomcinoddgooolagloogehp)
2. Open the stackblitz
3. Run the accessibility checker
4. Filter on violations only and note the errors: 'The 'id' "pn_id_363_source_list_list" specified for the ARIA property 'aria-owns' value is not valid' and 'The 'id' "pn_id_363_target_list_list" specified for the ARIA property 'aria-owns' value is not valid'

### Expected behavior

There are no accessibility violations for the p-pickList component
